frontend/packages/console-app/src/components/quick-starts/utils/useQuickStartPermission.ts

```diff
@@ -0,0 +1,29 @@
+import { useAccessReview } from '@console/internal/components/utils';
+import { useFlag } from '@console/shared/src/hooks/flag';
+import { QuickStart } from './quick-start-types';
+
+const useQuickStartPermission = (quickStart: QuickStart): boolean => {
+  const {
+    spec: { accessReviewResources, flags },
+  } = quickStart;
+
+  const hasAccess =
+    !accessReviewResources ||
+    // access review resources for a specific quick start is going to be static so the order will not change
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    accessReviewResources.map((descriptor) => useAccessReview(descriptor)).every((x) => x);
+
+  const requiredFlags =
+    !flags?.required ||
+    // flags for a specific quick start is going to be static so the order will not change
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    flags?.required?.map((flag) => useFlag(flag)).every((x) => x);
+
+  // flags for a specific quick start is going to be static so the order will not change
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const disallowedFlags = flags?.disallowed?.map((flag) => useFlag(flag)).some((x) => x);
+
+  return hasAccess && requiredFlags && !disallowedFlags;
```

Comment:
Shouldn't you distinguish between access review loading and loaded?
Otherwise you may be replying that access is not yet available while the review is still pending.
`useAccessReview2` returns a tuple with a loading indicator.

Reply:
Done.